SourceX/dvlnet/tcp_server.cpp

```diff
@@ -82,6 +82,7 @@ void tcp_server::handle_recv(scc con, const asio::error_code &ec,
 				handle_recv_packet(*pkt);
 			}
 		} catch (dvlnet_exception &e) {
+			SDL_Log("Network error");

```

Comment:
`"Network error: %s", e.what()`

Reply:
```
/home/ajenbo/code/diablo/devilutionX/SourceX/dvlnet/tcp_server.cpp:85:40: error: ‘virtual const char* dvl::net::dvlnet_exception::what() const’ is private within this context
   85 |    SDL_Log("Network error: %s", e.what());
      |                                        ^

```

Reply:
`what` is public in the base class so its visiblity should not be narrowed by descendant classes https://en.cppreference.com/w/cpp/error/exception/what

We should fix this in `dvlnet_exception`

Reply:
Sorry I have no idea how this work in C++, I tried changing it to `virtual const char *what() const throw()` but it didn't help. I will probably need to expclicity need you to point out the needed change :'D